### PR TITLE
Fix: make prover not certify transactions stored but not certified yet

### PR DESCRIPTION
**Pull request by jpraynaud**
## Content
This PR includes a fix to the **Cardano transactions prover** to avoid certifying transactions stored in the database, but not already certified. This was causing the error: `Proof and certificate don't match` when trying to verify this type of transactions instead of marking them as **not certified**.

### Benchmarks

Screenshot from 2024-06-06 15-23-44

| Total Requests | Concurrency | Transactions/Request | Pool (x50) : Request/s | Pool (x50) + Fix not yet certified transactions: Request/s |
| -------------- | ----------- | -------------------- | ---------------------- | ---------------------------------------------------------- |
| 1000           | 100         | 1                    | **273.78**                 | **275.95**                                                     |
| 1000           | 100         | 2                    | **201.6**                  | **200.65**                                                     |
| 1000           | 100         | 3                    | **143.14**                 | **150.64**                                                     |
| 1000           | 100         | 4                    | **115.41**                 | **124.02**                                                     |
| 1000           | 100         | 5                    | **91.15**                  | **103.31**                                                     |
| 1000           | 100         | 6                    | **80.58**                  | **87.72**                                                      |
| 1000           | 100         | 7                    | **74.23**                  | **77.33**                                                      |
| 1000           | 100         | 8                    | **65.05**                  | **68.08**                                                      |
| 1000           | 100         | 9                    | **58.78**                  | **61.77**                                                      |
| 1000           | 100         | 10                   | **52.51**                  | **58.13**                                                      |

> Running on **Linux / 8 cores / 64 GB RAM / 2 TB SSD**

## Pre-submit checklist

- Branch
  - [x] Tests are provided (if possible)
  - [x] Crates versions are updated (if relevant)
  - [x] Commit sequence broadly makes sense
  - [x] Key commits have useful messages
- PR
  - [x] No clippy warnings in the CI
  - [x] Self-reviewed the diff
  - [x] Useful pull request description
  - [x] Reviewer requested

## Issue(s)
Relates to #1719

### CODE REVIEW

File: mithril-aggregator/src/services/prover.rs [hunk 1]
```diff
@@ -410,6 +412,50 @@ mod tests {
         transactions_set_proof[0].verify().unwrap();
     }
 
+    #[tokio::test]
+    async fn cant_compute_proof_for_not_yet_certified_transaction() {
+        let total_block_ranges = 5;
```
Comment (sfauvel) [inline mithril-aggregator/src/services/prover.rs:417]: Is it useful to declare a variable here ?  
Builder function names seem to gave enough information about the value.

**Review by sfauvel — APPROVED** [review]
LGTM

File: internal/mithril-persistence/src/database/repository/cardano_transaction_repository.rs [hunk 2]
```diff
@@ -327,13 +329,14 @@ mod tests {
                 CardanoTransactionRecord::new("tx_hash-123", 10, 50, "block_hash-123", 99),
                 CardanoTransactionRecord::new("tx_hash-456", 11, 51, "block_hash-456", 100),
```
[the remark this answers is not in the document]
Comment (jpraynaud) [reply internal/mithril-persistence/src/database/repository/cardano_transaction_repository.rs:330]: Done

File: mithril-aggregator/src/services/prover.rs (hunk 1 above)
Comment (jpraynaud) [reply mithril-aggregator/src/services/prover.rs:417]: Actually, all the tests in the module are structured this way. 
But I agree we can skip the variables as the function names are clear enough.

Comment (jpraynaud) [reply mithril-aggregator/src/services/prover.rs:417]: Done

**Review by Alenar — APPROVED** [review]

File: mithril-aggregator/src/services/prover.rs [hunk 3]
```diff
@@ -410,6 +412,50 @@ mod tests {
...
+        let total_transactions_per_block_range = 3;
```
Comment (dlachaume) [inline mithril-aggregator/src/services/prover.rs]: If the variable is kept, it should be renamed as follows
```suggestion
        let blocks_per_block_range = 3;
```

**Review by dlachaume — APPROVED** [review]

**Pull request merged**